app/src/app/utils/events/mapEvents.ts

```diff
@@ -0,0 +1,175 @@
+    } else if (activeTool === "eraser") {
+      // erase features
+      // TODO: implement eraser
+    }
+  } else {
+    // tbd, for pan mode - is there an info mode on click?
+  }
+};
+
+export const handleMapMouseUp = (
+  e: MapLayerMouseEvent | MapLayerTouchEvent,
+  map: MutableRefObject<Map | null>,
+  hoverFeatureIds: React.MutableRefObject<Set<string>>
+) => {
+  const mapStore = useMapStore.getState();
+  const activeTool = mapStore.activeTool;
+  const isPainting = mapStore.isPainting;
+
+  if (activeTool === "brush" && isPainting) {
+    // set isPainting to false
+    mapStore.setIsPainting(false);
+  }
+};
+
+export const handleMapMouseDown = (
+  e: MapLayerMouseEvent | MapLayerTouchEvent,
+  map: MutableRefObject<Map | null>,
+  hoverFeatureIds: React.MutableRefObject<Set<string>>
+) => {
+  const mapStore = useMapStore.getState();
+  const activeTool = mapStore.activeTool;
+
+  if (activeTool === "pan") {
+    return;
+  } else if (activeTool === "brush" || activeTool === "eraser") {
+    // disable drag pan
+    map.current?.dragPan.disable();
+    if (activeTool === "brush") {
+      mapStore.setIsPainting(true);
+      return;
+    } else if (activeTool === "eraser") {
+      // erase features tbd
+    }
+  }
+};
+
+export const handleMapMouseEnter = (
+  e: MapLayerMouseEvent | MapLayerTouchEvent,
+  map: MutableRefObject<Map | null>,
+  hoverFeatureIds: React.MutableRefObject<Set<string>>
+) => {};
+
+export const handleMapMouseOver = (
+  e: MapLayerMouseEvent | MapLayerTouchEvent,
+  map: MutableRefObject<Map | null>,
+  hoverFeatureIds: React.MutableRefObject<Set<string>>
+) => {};
+
+export const handleMapMouseLeave = (
+  e: MapLayerMouseEvent | MapLayerTouchEvent,
+  map: MutableRefObject<Map | null>,
+  hoverFeatureIds: React.MutableRefObject<Set<string>>
+) => {};
+
+export const handleMapMouseOut = (
+  e: MapLayerMouseEvent | MapLayerTouchEvent,
+  map: MutableRefObject<Map | null>,
+  hoverFeatureIds: React.MutableRefObject<Set<string>>
+) => {};
+
+export const handleMapMouseMove = (
+  e: MapLayerMouseEvent | MapLayerTouchEvent,
+  map: MutableRefObject<Map | null>,
+  hoverFeatureIds: React.MutableRefObject<Set<string>>
+) => {
+  const mapStore = useMapStore.getState();
+  const activeTool = mapStore.activeTool;
+  const isPainting = mapStore.isPainting;
+  const brushSize = mapStore.brushSize;
+  const bbox = boxAroundPoint(e, brushSize);
+  const selectedFeatures = map.current?.queryRenderedFeatures(bbox, {
+    layers: [BLOCK_LAYER_ID],
+  });
+  if (!isPainting) {
+    HighlightFeature(selectedFeatures, map, hoverFeatureIds);
+  } else if (activeTool === "brush" && isPainting) {
+    /**
+     * @todo
+     * what we really want is to set map feature state here,
+     * and then update the store with the new assignments when
+     * we mouseup, to avoid unnecessary rerenders and state updates.
+     * this should reduce the bottleneck from debouncing
+     * */
```

Comment:
Ooo nice I like that

Reply:
referred to in #7, will treat that as a discrete bit of work after merging this in